app/parse.py

```diff
@@ -1,12 +1,156 @@
-from dataclasses import dataclass
+import csv
+import time
+
+import requests
+from selenium import webdriver
+from selenium.webdriver.common.by import By
+from dataclasses import dataclass, fields
+from bs4 import BeautifulSoup
+
+BASE_URL = "https://mate.academy"
+
+
+class Driver:
+    instance = None
+
+    @staticmethod
```

Comment:
This method is a singleton implementation for WebDriver instance. It's a good practice to manage WebDriver instances carefully, but this implementation has a few issues. First, it doesn't handle WebDriver termination. When you're done with the WebDriver, it's a good practice to call `driver.quit()` to ensure that the underlying WebDriver instance is properly closed. This is especially important when you're using a WebDriver in a long-running application, as not properly closing WebDriver can lead to resource leaks.